xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/internal/job/AbstractDeletedDocumentsJob.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.refactoring.internal.job;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.xwiki.job.AbstractJob;
+import org.xwiki.job.AbstractJobStatus;
+import org.xwiki.job.GroupedJob;
+import org.xwiki.job.JobGroupPath;
+import org.xwiki.job.Request;
+import org.xwiki.model.ModelContext;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.WikiReference;
+import org.xwiki.refactoring.internal.ModelBridge;
+import org.xwiki.refactoring.job.AbstractDeletedDocumentsRequest;
+import org.xwiki.refactoring.job.RefactoringJobs;
+
+/**
+ * Define an abstract job that handles deleted documents.
+ * Typically this class has been created to abstract mechanism of {@link RestoreJob} and {@link PermanentlyDeleteJob}.
+ *
+ * @version $Id$
+ * @since 10.10RC1
+ */
+public abstract class AbstractDeletedDocumentsJob extends AbstractJob<AbstractDeletedDocumentsRequest,
+    AbstractJobStatus<AbstractDeletedDocumentsRequest>> implements GroupedJob
+{
+    static final JobGroupPath ROOT_GROUP = new JobGroupPath(RefactoringJobs.GROUP, null);
+
+    @Inject
+    protected ModelBridge modelBridge;
+
+    @Inject
+    protected ModelContext modelContext;
+
+    private JobGroupPath groupPath;
+
+    @Override
+    protected AbstractJobStatus<AbstractDeletedDocumentsRequest> createNewStatus(
+        AbstractDeletedDocumentsRequest request)
+    {
+        AbstractJobStatus<AbstractDeletedDocumentsRequest> status = super.createNewStatus(request);
+
+        status.setCancelable(true);
```

Comment:
Seems to me that would make more sense in AbstractDeletedDocumentsRequest constructor for example.